packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,193 @@
+  }
+  return props;
+};
+
+const getFlooredFixed = (value: number, precision: number): number => {
+  const factor = 10 ** precision;
+  const roundedValue = Math.floor(value * factor) / factor;
+  return Number(roundedValue.toFixed(precision));
+};
+
+const addCommas = (num: number, currency: currencyType): string => {
+  const locale = currencylocaleMapping[currency];
+  return num.toLocaleString(locale);
+};
+
+const getFormattedAmountWithSuffixSymbol = (num: number, currency: currencyType): string => {
+  const abbreviations: { value: number; symbol: string }[] = currencyAbbreviationsMapping[currency];
+
+  const abbreviation = abbreviations.find((abbr) => num >= abbr.value);
+  if (abbreviation) {
+    num = num / abbreviation.value;
+    const formattedNum = Number(getFlooredFixed(num, 2));
+    return addCommas(formattedNum, currency) + abbreviation.symbol;
+  } else {
+    return num.toString();
+  }
+};
+
+const formatAmountWithSuffix = (
+  suffix: AmountProps['suffix'],
+  num: number,
+  currency: currencyType,
+): string => {
+  switch (suffix) {
+    case suffixTypes.DECIMALS: {
+      const decimalNum = getFlooredFixed(num, 2);
+      return addCommas(decimalNum, currency);
+    }
+    case suffixTypes.HUMANIZE: {
+      return getFormattedAmountWithSuffixSymbol(num, currency);
+    }
+    default:
+      return addCommas(getFlooredFixed(num, 0), currency);
+  }
+};
+
+const getCurrencyWeight = (
+  isAffixSubtle: NonNullable<AmountProps['isAffixSubtle']>,
+  size: NonNullable<AmountProps['size']>,
+): 'bold' | 'regular' => {
+  if (isAffixSubtle || size.includes('bold')) return 'bold';
+  return 'regular';
+};
+
+const Amount = ({
+  value,
+  suffix = 'decimals',
+  size = 'heading-small',
+  isAffixSubtle = true,
+  intent,
+  prefix = 'currency-symbol',
+  testID,
+}: AmountProps): ReactElement => {
+  if (typeof value !== 'number' && !isNaN(value)) {
+    throw new Error('[Blade: Amount]: Number as value is required for Amount.');
+  }
+
+  // This will be added to prop and can be switched to a currency
+  const currency = 'INR';
+  const currencyPrefix = currencyPrefixMapping[currency][prefix];
+  const renderedValue = formatAmountWithSuffix(suffix, value, currency);
+  const { textColor, prefixSuffixColor } = getColorProps({
+    intent,
+  });
+
+  const currencyColor = isAffixSubtle ? prefixSuffixColor : textColor;
+  const currencyFontSize = getAffixFontSize(isAffixSubtle, size);
+  const currencyWeight = getCurrencyWeight(isAffixSubtle, size);
+
+  return (
+    <StyledAmount {...metaAttribute({ name: MetaConstants.Amount, testID })}>
+      <BaseBox
+        display="flex"
+        alignItems="baseline"
+        paddingLeft="spacing.2"
+        paddingRight="spacing.2"
+      >
+        <BaseBox paddingRight="spacing.1">
+          <BaseText fontWeight={currencyWeight} fontSize={currencyFontSize} color={currencyColor}>
+            {currencyPrefix}
+          </BaseText>
+        </BaseBox>
+        <AmountValue
```

Comment:
Why do we need a separate file and component for `AmountValue`? Its not re-used and neither does it have native/web specific things. Lets render everything in `Amount.tsx` directly only and remove `AmountValue` component?